src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java

```diff
@@ -68,40 +84,10 @@ protected void dig(Block b, BlockMenu menu, Block block) {
     }
 
     @Override
+    @ParametersAreNonnullByDefault
     protected void moveAndDig(Block b, BlockMenu menu, BlockFace face, Block block) {
-        Collection<ItemStack> drops = block.getDrops(effectivePickaxe);
-
-        if (!SlimefunTag.UNBREAKABLE_MATERIALS.isTagged(block.getType()) && !drops.isEmpty()) {
-            OfflinePlayer owner = Bukkit.getOfflinePlayer(UUID.fromString(BlockStorage.getLocationInfo(b.getLocation(), "owner")));
-
-            if (SlimefunPlugin.getProtectionManager().hasPermission(owner, block.getLocation(), ProtectableAction.BREAK_BLOCK)) {
-                AndroidMineEvent event = new AndroidMineEvent(block, new AndroidInstance(this, b));
-                Bukkit.getPluginManager().callEvent(event);
-
-                if (event.isCancelled()) {
-                    return;
-                }
-
-                // We only want to break non-Slimefun blocks
-                SlimefunItem blockId = BlockStorage.check(block);
-                if (blockId == null) {
-                    for (ItemStack drop : drops) {
-                        if (menu.fits(drop, getOutputSlots())) {
-                            menu.pushItem(drop, getOutputSlots());
-                        }
-                    }
-
-                    block.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, block.getType());
-
-                    block.setType(Material.AIR);
-                    move(b, face, block);
-                }
-            } else {
-                move(b, face, block);
-            }
-        } else {
-            move(b, face, block);
-        }
+        dig(b, menu, block);
+        move(b, face, block);
```

Comment:
I don't think this simplification works here because it would be quite different.